src/token.rs

```diff
@@ -51,7 +54,16 @@ impl Tokenizer {
         }
 
         if string.starts_with("--") {
-            return Some(Token::Long(&string[2..]));
+            return match string.find('=') {
+                // "--foo bar" case
+                None => Some(Token::Long(&string[2..])),
+                // "--foo-bar" case, save value in self.pending
+                Some(i) => {
+                    self.pending = &string[i + 1..];
+                    self.pending_index = 0;
```

Comment:
If foo does not expect an arg, it looks like this would end up treating `--foo=bar` as if it were `--foo -b -a -r` which seems wrong. Could you make sure that doesn't happen?

Reply:
Done.

One wrinkle is that `--foo= --some-other-flag` (note the space after `=`) will be treated as `--foo=--some-other-flag`. It still gets reported as an error though, assuming that `--foo` is not a boolean.

Reply:
I did think about returning an error here instead -- if we see `--foo=` (i.e., there's no arg) then it doesn't matter what the flag spec is, this should be an error.

But there's no mechanism at the moment for the tokenizer to return an error, so I figured this was better kept with the error handling in `parser.rs`.